ga4gh/protocol.py

```diff
@@ -56,39 +57,15 @@ def __eq__(self, other):
         TODO This feature is experimental and requires more testing; use
         with caution!!
         """
-        # TODO This implementation is ugly! There must be a nicer way to do
-        # this...
-        ret = False
-        if type(other) == type(self):
-            ret = True
-            for f in self.schema.fields:
-                k = f.name
-                a1 = getattr(self, k)
-                a2 = getattr(other, k)
-                if self.isEmbeddedType(k):
-                    if isinstance(f.type, avro.schema.ArraySchema):
-                        if len(a1) == len(a2):
-                            ret = ret and all(x == y for x, y in zip(a1, a2))
-                        else:
-                            ret = False
-                    else:
-                        ret = ret and a1 == a2
-                else:
-                    ret = ret and a1 == a2
-        return ret
+        if type(other) != type(self):
+            return False
+
+        fieldNames = imap(lambda f: f.name, self.schema.fields)
```

Comment:
I would argue that lambdas are a special case given they extremely limited syntactic scoping. I’ll leave this around for comment though.

On Monday, January 12, 2015 at 9:44 AM, Jerome Kelleher wrote:

> In ga4gh/protocol.py (https://github.com/ga4gh/server/pull/64#discussion-diff-22793383):
> > - a2 = getattr(other, k) > - if self.isEmbeddedType(k): > - if isinstance(f.type, avro.schema.ArraySchema): > - if len(a1) == len(a2): > - ret = ret and all(x == y for x, y in zip(a1, a2)) > - else: > - ret = False > - else: > - ret = ret and a1 == a2 > - else: > - ret = ret and a1 == a2 > - return ret > + if type(other) != type(self): > + return False > + > + fieldNames = imap(lambda f: f.name (http://f.name), self.schema.fields)  
> f should probably be field here if we're going to be consistent. I'm comfortable with f, but I don't know where we draw the line. @kerrydc (https://github.com/kerrydc), what's your position?
>  
> —
> Reply to this email directly or view it on GitHub (https://github.com/ga4gh/server/pull/64/files#r22793383).
>  
>  
>  
>  
>